cms/templates/textbooks.html

```diff
@@ -31,6 +31,11 @@
   <%static:invoke_page_bundle page_name="js/pages/textbooks" class_name="TextbooksFactory">
      ${textbooks | n, dump_js_escaped_json}
   </%static:invoke_page_bundle>
+  ${static.renderReact(
+    component="WrappedNoTextbooks",
+    id="no-textbooks-container",
+    store="StudioStore"
+  )}
```

Comment:
Note how here we include the `renderReact` call inside the `page_bundle` block, AFTER `static:invoke_page_bundle`. This is because `StudioStore` depends on the collection which will be initialized by `TextbooksFactory`.

Reply:
I don't believe team Dahlia has used renderReact yet for our studio-frontend work in edx-platform, but I imagine we'll have to leverage it for the in-context image selection work, since the React component will be in a Backbone page, right? Just so that I'm understanding correctly, we will also pass the StudioStore to the script, since that's the universal Redux store for studio in edx-platform. I'm having a little trouble understanding how we can interface with StudioStore from studio-frontend. Right now the entire Redux store for our files & uploads page lives within studio-frontend, and the store is initialized there as well. I'm not sure how that would work in cases where we'd need to interface with Backbone, since the store is being initialized/passed to the React component in edx-platform. Would we write action creators and reducers in studio-frontend, but the actual store would live in edx-platform? Or something else entirely?

If it's easier to explain in person, I'd love to chat tomorrow, since I'm working from home today (provided you don't have to merge it ASAP or anything; if so, no worries).

Reply:
We discussed this a bit in FedX but I wanted to get some thoughts down in here as well. The way I see it here are your options:
- Use SFE's own store only. Set the Backbone collection to a global location (`window.models.myModel`) and use that to hydrate the state of SFE's store, the same way we do with the course object. This would require setting up backbone-redux in studio-frontend. This is nice because the whole data layer is encapsulated in SFE, but annoying cause it means pulling backbone ugliness into SFE's new-and-shiny.
- Use Studio's store only. With this implementation, you'd export an un-wrapped modal component for SFE's production build and invoke it with `renderReact`, passing in `StudioStore` as this PR has done. This breaks the nice encapsulation/decoupling of SFE -- but considering this piece of UI is heavily dependent on existing UI, it's not really all that decoupled in the first place.
- Somehow pass data from Studio's store into SFE's store. This seems like probably the cleanest/best practice approach, but it has not been explored as part of this PR.
- Give the Backbone code access to SFE's store, and call `syncCollections` at the point in the legacy code where the collection is initialized (in lieu of https://github.com/edx/edx-platform/pull/17641/files#diff-864d9669b071699b5baf0c50622671deR8). I'm also not sure whether this is feasible, but if the Backbone file is being fed through webpack you should be able to import es2015 modules into it.

I'd lean towards option 2, but it depends on your use case.